time.c

```diff
@@ -1888,39 +1888,25 @@ force_make_tm(VALUE time, struct time_object *tobj)
 }
 
 static void
-time_mark(void *ptr)
+time_mark_and_move(void *ptr)
 {
     struct time_object *tobj = ptr;
     if (!FIXWV_P(tobj->timew)) {
-        rb_gc_mark_movable(w2v(tobj->timew));
+        rb_gc_mark_and_move(&WIDEVAL_GET(tobj->timew));
```

Comment:
I feel like this still needs to use `w2v` in some way, previously there was https://github.com/ruby/ruby/commit/b8de3cfb04e4d510d3a3af73029a9ca5ab39f79e. On some architecures, I am getting the following build error:

```
time.c: In function 'time_mark_and_move':
time.c:1895:29: error: passing argument 1 of 'rb_gc_mark_and_move' from incompatible pointer type [-Wincompatible-pointer-types]
1895 |         rb_gc_mark_and_move(&WIDEVAL_GET(tobj->timew));
|                             ^
|                             |
|                             wideval_t * {aka long long unsigned int *}
In file included from ./shape.h:4,
from internal/variable.h:16,
from time.c:43:
./internal/gc.h:213:33: note: expected 'VALUE *' {aka 'long unsigned int *'} but argument is of type 'wideval_t *' {aka 'long long unsigned int *'}
213 | void rb_gc_mark_and_move(VALUE *ptr);
|                          ~~~~~~~^~~
```

Maybe so?

```suggestion
        VALUE timew = w2v(tobj->timew);
        rb_gc_mark_and_move(&timew);
```

Reply:
👀